reach4help/reach4help
# Feat: Backend Functions to Delete User Data

## Description
Functionality to delete a user account and its associated data (requests, privilegedInfo, timelines)
- backend function to delete a user's privilegedInformation sub-collection
- backend function to 'delete' all requests associated with user for both PIN and CAV
- backend function to 'delete' timelines associated with user
- combined HTTPS callable cloud function to delete all user data listed above and its action creator in the frontend
## Motivation
Implement ability for a user to delete his/her account
## Testing Guidelines
Call deleteUserData() action, check if requests, privilegedInfo, timelines associated with user has been deleted
## Release Checklist

- [y] I have a plan for how to revert, rollback, or disable these changes if things are not working as expected

## Security Checklist

This PR creates, modifies, or deletes:

- [y] User data: personal information handling, logs, error messages, etc.



## Additional Notes
